src/windows/wslaservice/exe/WSLAVirtualMachine.cpp

```diff
@@ -513,15 +530,62 @@ void WSLAVirtualMachine::ConfigureNetworking()
             wil::unique_socket{(SOCKET)process->GetStdHandle(gnsChannelFd).release()},
             config,
             dnsChannelFd != -1 ? wil::unique_socket{(SOCKET)process->GetStdHandle(dnsChannelFd).release()} : wil::unique_socket{});
+    }
+    else if (m_settings.NetworkingMode == WSLANetworkingModeVirtioProxy)
+    {
+        // Launch GNS
+        std::vector<WSLA_PROCESS_FD> fds(1);
+        fds[0].Fd = -1;
+        fds[0].Type = WSLAFdType::WSLAFdTypeDefault;
+
+        std::vector<const char*> cmd{"/gns", LX_INIT_GNS_SOCKET_ARG};
 
-        m_networkEngine->Initialize();
+        // If DNS tunnelling is enabled, use an additional for its channel.
+        THROW_HR_IF_MSG(E_NOTIMPL, m_settings.EnableDnsTunneling, "DNS tunneling not currently supported for VirtioProxy");
+
+        WSLA_PROCESS_OPTIONS options{};
+        options.Executable = "/init";
+        options.Fds = fds.data();
+        options.FdsCount = static_cast<DWORD>(fds.size());
+
+        // Because the file descriptors numbers aren't known in advance, the command line needs to be generated after the file
+        // descriptors are allocated.
+
+        std::string socketFdArg;
+        int gnsChannelFd = -1;
+        auto prepareCommandLine = [&](const auto& sockets) {
```

Comment:
Ah I should wire this in the ProcessLauncher util, so we can use it here too

Reply:
I moved things around a bit more to share the create process stuff so this will be easier to do.